common/tools/mcp-server/src/tools/listPackages.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { z } from "zod";
+import { readFile } from "node:fs/promises";
+import path, { dirname } from "node:path";
+import stripJsonComments from "strip-json-comments";
+import { fileURLToPath } from "node:url";
+import { CallToolResult } from "@modelcontextprotocol/sdk/types.js";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+let _projectSchema: ListPackagesSchema | undefined = undefined;
+
+async function getProjectSchema(): Promise<ListPackagesSchema> {
+  if (_projectSchema) return _projectSchema;
+
+  const rushJsonText = await readFile(path.resolve(__dirname, "../../../../../rush.json"), "utf-8");
```

Comment:
should the root of the repo be an argument to the server/tool? also, I believe we have some good APIs for parsing the repo structure in dev-tool, wonder if they're easily used here too.